src/preload.py

```diff
@@ -913,6 +913,7 @@ def main_preload(app_data, additional_bytes, splash_image_path):
 
 
 def get_image_info():
+    PARTITIONS_CACHE[None] = prepare_global_partitions()
```

Comment:
My unease here, if I am reading it right, is that `get_image_info()` and `main_preload()` are two top-level "exported-so-to-speak" methods:
```
methods = {
    "get_image_info": get_image_info,
    "preload": main_preload,
}

if __name__ == "__main__":
    ...
    for line in sys.stdin:
        try:
            data = json.loads(line)
            method = methods[data["command"]]
            result = method(**data.get("parameters", {}))
```

Both methods eventually call `get_partition()`, but only `get_image_info()` initializes `PARTITIONS_CACHE[None]` with this PR's proposed changes. If `main_preload()` was called before or instead of `get_image_info()`, `PARTITIONS_CACHE[None]` would go uninitialized.

For want of a better idea, I could suggest moving `PARTITIONS_CACHE[None] = prepare_global_partitions()` to some top-level `init()` method that both `get_image_info()` and `main_preload()` would call, subject to some "is_initialized" flag to avoid calling `init()` more than once.